src/main/cli/cli.c

```diff
@@ -4162,176 +4095,188 @@ STATIC_UNIT_TESTED void cliSet(char *cmdline)
         cliPrintLine("Current settings: ");
 
         for (uint32_t i = 0; i < valueTableEntryCount; i++) {
-            const clivalue_t *val = &valueTable[i];
-            cliPrintf("%s = ", valueTable[i].name);
-            cliPrintVar(val, len); // when len is 1 (when * is passed as argument), it will print min/max values as well, for gui
-            cliPrintLinefeed();
+            cliPrintValueTableItem(i);
         }
-    } else if ((eqptr = strstr(cmdline, "=")) != NULL) {
-        // has equals
+        return;
+    } 
+    
+    if ((eqptr = strstr(cmdline, "=")) == NULL) {
+        // no equals, check for matching variables.
+        cliGet(cmdline);
+        return;
+    }
 
-        uint8_t variableNameLength = getWordLength(cmdline, eqptr);
+    // has equals
+    uint8_t variableNameLength = getWordLength(cmdline, eqptr);
 
-        // skip the '=' and any ' ' characters
-        eqptr++;
-        eqptr = skipSpace(eqptr);
+    // skip the '=' and any ' ' characters
+    eqptr++;
+    eqptr = skipSpace(eqptr);
 
-        const uint16_t index = cliGetSettingIndex(cmdline, variableNameLength);
-        if (index >= valueTableEntryCount) {
-            cliPrintErrorLinef("INVALID NAME");
-            return;
+    uint8_t settingIndex = 0;
+
+    // need to hunt for the [x] for the index - simple use case here.
+    if (variableNameLength > 4 && cmdline[variableNameLength-1] == ']') {
+        if (cmdline[variableNameLength-3] == '[') {
```

Comment:
We could get arrays > 99 at some point - should we do this with an `atoi()`?

Also, coding standard says there should be spaces around operators.

Reply:
Spaces around operators added.

If and when we have more than 99 in an array - it can be dealt with. Simply a case of YAGNI. In reality if we have an array > 99 we have a problem.